src/frontend/entry-button.c

```diff
@@ -298,10 +300,73 @@ static void brisk_menu_entry_drag_data(GtkWidget *widget, __brisk_unused__ GdkDr
         gtk_selection_data_set_uris(data, (gchar **)uris);
 }
 
-static void brisk_menu_entry_clicked(GtkButton *widget)
+static void build_menu(__brisk_unused__ gpointer key, gpointer value, gpointer user_data)
+{
+        BriskBackend *backend = BRISK_BACKEND(value);
+        BriskMenuEntryButton *self = BRISK_MENU_ENTRY_BUTTON(user_data);
+        GSList *actions = brisk_backend_get_item_actions(backend, self->item);
+
+        if (actions == NULL) {
+                return;
+        }
+
+        if (g_list_length(gtk_container_get_children(GTK_CONTAINER(self->menu))) != 0 &&
+            g_slist_length(actions) != 0) {
+                GtkWidget *separator = gtk_separator_menu_item_new();
+                gtk_menu_shell_append(GTK_MENU_SHELL(self->menu), separator);
+                gtk_widget_show(GTK_WIDGET(separator));
+        }
+
+        guint i;
+        GSList *node;
+
+        for (i = 0; (node = g_slist_nth(actions, i)); i++) {
+                GtkWidget *menu_item =
+                    gtk_menu_item_new_with_label(g_action_get_name(G_ACTION(node->data)));
+                gtk_menu_shell_append(GTK_MENU_SHELL(self->menu), menu_item);
+                gtk_widget_show(GTK_WIDGET(menu_item));
+                g_signal_connect_swapped(menu_item,
+                                         "activate",
+                                         G_CALLBACK(brisk_menu_entry_menu_item_activated),
+                                         node->data);
+        }
+}
+
+static gboolean brisk_menu_entry_button_release_event(GtkWidget *widget,
+                                                      GdkEventButton *event_button)
 {
         BriskMenuEntryButton *self = BRISK_MENU_ENTRY_BUTTON(widget);
-        brisk_menu_entry_button_launch(self);
+
+        if (event_button->button == GDK_BUTTON_PRIMARY) {
```

Comment:
I don't like how this flows, the `else if` is just unnecessary. Handle the primary, then if the button **doesn't match** the secondary, just return in these spots on the parent method. Saves a whole bunch of indentation